plugins/extractors/bigquery/bigquery.go

```diff
@@ -257,10 +257,37 @@ func (e *Extractor) buildAsset(ctx context.Context, t *bigquery.Table, md *bigqu
 	tableURN := plugins.BigQueryURN(t.ProjectID, t.DatasetID, t.TableID)
 
 	tableProfile := e.buildTableProfile(tableURN, tableStats)
-
 	var partitionField string
+	partitionData := make(map[string]interface{}, 0)
 	if md.TimePartitioning != nil {
-		partitionField = md.TimePartitioning.Field
+		timePartitionMap := make(map[string]interface{}, 0)
+		if md.TimePartitioning.Field != "" {
+			partitionField = md.TimePartitioning.Field
+		} else {
+			partitionField = "_PARTITIONTIME"
+		}
+		timePartitionMap["partition_by"] = string(md.TimePartitioning.Type)
+		timePartitionMap["partition_expire"] = md.TimePartitioning.Expiration.Seconds()
```

Comment:
Should we indicate in the field name that the unit is seconds?

Reply:
Yes that would be good. Will add.

Reply:
Done